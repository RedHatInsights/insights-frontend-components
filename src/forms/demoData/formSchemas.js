export const simple = {
    title: 'A registration form',
    description: 'A simple form example.',
    type: 'object',
    required: [
        'firstName',
        'lastName'
    ],
    properties: {
        firstName: {
            type: 'string',
            title: 'First name'
        },
        lastName: {
            type: 'string',
            title: 'Last name'
        },
        age: {
            type: 'integer',
            title: 'Age'
        },
        bio: {
            type: 'string',
            title: 'Bio'
        },
        password: {
            type: 'string',
            title: 'Password',
            minLength: 3
        },
        telephone: {
            type: 'string',
            title: 'Telephone',
            minLength: 10
        }
    }
};

export const uiSchemaSimple = {
    firstName: {
        'ui:autofocus': true,
        'ui:emptyValue': ''
    },
    age: {
        'ui:widget': 'updown',
        'ui:title': 'Age of person',
        'ui:description': '(earthian year)'
    },
    bio: {
        'ui:widget': 'textarea'
    },
    password: {
        'ui:widget': 'password',
        'ui:help': 'Hint: Make it strong!'
    },
    telephone: {
        'ui:options': {
            inputType: 'tel'
        }
    }
};

/**
 * Lauras form usecases
 */

export const lauraSchema1 = {
    type: 'object',
    required: [ 'title' ],
    properties: {
        title: { type: 'string', title: 'Title', default: 'A new provider' },
        done: { type: 'boolean', title: 'Done?', default: false }
    }
};

export const lauraSchema2 = {
    title: 'Add an Openshift Provider',
    type: 'object',
    properties: {
        name: { title: 'Provider Name', type: 'string' },
        description: { title: 'Description', type: 'string' },
        url: { title: 'URL', type: 'string' },
        verify_ssl: { title: 'Verify SSL', type: 'boolean', default: false }, // eslint-disable-line camelcase
        user: { title: 'User Name', type: 'string', default: '' },
        token: { title: 'Token', type: 'string', default: '' },
        password: { title: 'Password', type: 'string', minlength: 6 }
    },
    required: [ 'name', 'url' ]
};

export const lauraUiSchema = {
    password: {
        'ui:widget': 'password'
    }
};

export const benchmarchUiSchema = {
    '0typnpcsapb': {
        'ui:widget': 'textarea'
    }

};

export const benchmark = {
    title: 'Add an Openshift Provider',
    type: 'object',
    required: [ 'name', 'url' ],
    properties: {
        '0typnpcsapb': { title: 'Provider Name', type: 'string' },
        '0ujxmbpheqp': { title: 'Description', type: 'string' },
        '0igxbbgefiy': { title: 'URL', type: 'string' },
        '0ymxsbbnvhr': { title: 'Verify SSL', type: 'boolean', default: false },
        '0sinrdrawrc': { title: 'Token', type: 'string', default: '' },
        '0pzhrzaqlev': { title: 'Provider Name', type: 'string' },
        '0grbumqfbdf': { title: 'Password', type: 'string', minlength: 6 },
        '1xbwrkocghx': { title: 'Provider Name', type: 'string' },
        '1jlnvrdletr': { title: 'Description', type: 'string' },
        '1nutqqzmmdl': { title: 'URL', type: 'string' },
        '1otfcpssbia': { title: 'Verify SSL', type: 'boolean', default: false },
        '1qoekvwtlxh': { title: 'Token', type: 'string', default: '' },
        '1bznwozzuel': { title: 'Provider Name', type: 'string' },
        '1uukcsggjcw': { title: 'Password', type: 'string', minlength: 6 },
        '2hbxarbsnwl': { title: 'Provider Name', type: 'string' },
        '2zrdrvfjfgn': { title: 'Description', type: 'string' },
        '2lmtpcpntqh': { title: 'URL', type: 'string' },
        '2tdrxrquqgh': { title: 'Verify SSL', type: 'boolean', default: false },
        '2roloctixhv': { title: 'Token', type: 'string', default: '' },
        '2rthxazlwbs': { title: 'Provider Name', type: 'string' },
        '2istklqsvxq': { title: 'Password', type: 'string', minlength: 6 },
        '3qbmuzkgmdd': { title: 'Provider Name', type: 'string' },
        '3qixbhoqwoa': { title: 'Description', type: 'string' },
        '3psewlqfnhw': { title: 'URL', type: 'string' },
        '3dzssevqlpb': { title: 'Verify SSL', type: 'boolean', default: false },
        '3mgaabkzwah': { title: 'Token', type: 'string', default: '' },
        '3kfbulbogxh': { title: 'Provider Name', type: 'string' },
        '3vmgsbhioqu': { title: 'Password', type: 'string', minlength: 6 },
        '4qpabiveask': { title: 'Provider Name', type: 'string' },
        '4jjnmcfdebp': { title: 'Description', type: 'string' },
        '4nbqouebuuk': { title: 'URL', type: 'string' },
        '4nfdevtvlrq': { title: 'Verify SSL', type: 'boolean', default: false },
        '4warabfuopr': { title: 'Token', type: 'string', default: '' },
        '4kxvvchpfeb': { title: 'Provider Name', type: 'string' },
        '4zovuggbhmp': { title: 'Password', type: 'string', minlength: 6 },
        '5xamdsohpad': { title: 'Provider Name', type: 'string' },
        '5nwfngvzrbh': { title: 'Description', type: 'string' },
        '5jsrsailgee': { title: 'URL', type: 'string' },
        '5whshmgodbt': { title: 'Verify SSL', type: 'boolean', default: false },
        '5pvvpckxzft': { title: 'Token', type: 'string', default: '' },
        '5nygowqcane': { title: 'Provider Name', type: 'string' },
        '5uevjqepatw': { title: 'Password', type: 'string', minlength: 6 },
        '6dpwkmlvdgt': { title: 'Provider Name', type: 'string' },
        '6tytrqcbetf': { title: 'Description', type: 'string' },
        '6kxxqxkedjl': { title: 'URL', type: 'string' },
        '6doprxmshog': { title: 'Verify SSL', type: 'boolean', default: false },
        '6zogeyyhleg': { title: 'Token', type: 'string', default: '' },
        '6imkionlcyk': { title: 'Provider Name', type: 'string' },
        '6fxxzhadqjn': { title: 'Password', type: 'string', minlength: 6 },
        '7uoxkyrrlzg': { title: 'Provider Name', type: 'string' },
        '7fqloslcdqm': { title: 'Description', type: 'string' },
        '7ysyghwfzda': { title: 'URL', type: 'string' },
        '7jpfvmjwqwt': { title: 'Verify SSL', type: 'boolean', default: false },
        '7usmfgpmeai': { title: 'Token', type: 'string', default: '' },
        '7tadcixfvfs': { title: 'Provider Name', type: 'string' },
        '7xmsikczfor': { title: 'Password', type: 'string', minlength: 6 },
        '8fahorsvlzj': { title: 'Provider Name', type: 'string' },
        '8vtptnyzxyx': { title: 'Description', type: 'string' },
        '8selkjedxna': { title: 'URL', type: 'string' },
        '8hziaaotznr': { title: 'Verify SSL', type: 'boolean', default: false },
        '8ypgtvdydfy': { title: 'Token', type: 'string', default: '' },
        '8pjfinvezwm': { title: 'Provider Name', type: 'string' },
        '8hxpuxrsotp': { title: 'Password', type: 'string', minlength: 6 },
        '9wlnegsvtzh': { title: 'Provider Name', type: 'string' },
        '9kgtpnqkjpu': { title: 'Description', type: 'string' },
        '9kozehumfdb': { title: 'URL', type: 'string' },
        '9wielcalalv': { title: 'Verify SSL', type: 'boolean', default: false },
        '9euladfeygy': { title: 'Token', type: 'string', default: '' },
        '9nsdbjfdhfo': { title: 'Provider Name', type: 'string' },
        '9ssxbbsmiey': { title: 'Password', type: 'string', minlength: 6 }
    }
};
