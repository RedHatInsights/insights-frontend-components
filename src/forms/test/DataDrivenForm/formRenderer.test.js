import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { lauraSchema2, lauraUiSchema, simple, uiSchemaSimple } from '../../demoData/formSchemas';
import FormRenderer from '../../DataDrivenForm/formRenderer';

describe('Form renderer', () => {
  it('should render simple form with input and switch', () => {
    const submit = jest.fn();
    const wrapper = mount(<FormRenderer schema={simple} uiSchema={uiSchemaSimple} onSubmit={submit} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not call submit function if form state is invalid', () => {
    const submit = jest.fn();
    const wrapper = mount(<FormRenderer schema={lauraSchema2} uiSchema={lauraUiSchema} onSubmit={submit} />);
    const submitButton = wrapper.find('button#form-renderer-submit');
    submitButton.simulate('click');
    expect(submit).not.toHaveBeenCalled();
  });

  it('should call submit function if form state valid', () => {
    const submit = jest.fn();
    const wrapper = mount(<FormRenderer schema={lauraSchema2} uiSchema={lauraUiSchema} onSubmit={submit} />);
    const submitButton = wrapper.find('button#form-renderer-submit');
    submitButton.simulate('click');
    expect(submit).not.toHaveBeenCalled();
    const nameInput = wrapper.find('input#name')
    nameInput.instance().value = 'Name';
    nameInput.simulate('change')

    const urlInput = wrapper.find('input#url');
    urlInput.instance().value = 'some.url';
    urlInput.simulate('change');

    submitButton.simulate('click');
    expect(submit).toHaveBeenCalledWith({ name: 'Name', url: 'some.url' }, expect.any(Object));
  });
});