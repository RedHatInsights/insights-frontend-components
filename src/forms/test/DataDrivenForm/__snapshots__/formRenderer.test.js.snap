// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renderer should render simple form with input and switch 1`] = `
<FormRenderer
  onSubmit={[MockFunction]}
  schema={
    Object {
      "description": "A simple form example.",
      "properties": Object {
        "age": Object {
          "title": "Age",
          "type": "integer",
        },
        "bio": Object {
          "title": "Bio",
          "type": "string",
        },
        "firstName": Object {
          "title": "First name",
          "type": "string",
        },
        "lastName": Object {
          "title": "Last name",
          "type": "string",
        },
        "password": Object {
          "minLength": 3,
          "title": "Password",
          "type": "string",
        },
        "telephone": Object {
          "minLength": 10,
          "title": "Telephone",
          "type": "string",
        },
      },
      "required": Array [
        "firstName",
        "lastName",
      ],
      "title": "A registration form",
      "type": "object",
    }
  }
  uiSchema={
    Object {
      "age": Object {
        "ui:description": "(earthian year)",
        "ui:title": "Age of person",
        "ui:widget": "updown",
      },
      "bio": Object {
        "ui:widget": "textarea",
      },
      "firstName": Object {
        "ui:autofocus": true,
        "ui:emptyValue": "",
      },
      "password": Object {
        "ui:help": "Hint: Make it strong!",
        "ui:widget": "password",
      },
      "telephone": Object {
        "ui:options": Object {
          "inputType": "tel",
        },
      },
    }
  }
>
  <div
    className="final-form-title"
  >
    <Title
      className=""
      size="3xl"
    >
      <h1
        className="pf-c-title pf-m-3xl"
      >
        A registration form
      </h1>
    </Title>
    <Title
      className=""
      size="lg"
    >
      <h1
        className="pf-c-title pf-m-lg"
      >
        A simple form example.
      </h1>
    </Title>
  </div>
  <ReactFinalForm
    initialValues={Object {}}
    onSubmit={[MockFunction]}
    render={[Function]}
    subscription={
      Object {
        "pristine": true,
        "submitting": true,
      }
    }
  >
    <form
      className=".pf-c-form"
    >
      <Grid
        className=""
        gutter={null}
        lg={null}
        md={null}
        sm={null}
        span={null}
        xl={null}
      >
        <div
          className="pf-l-grid"
        >
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={null}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <div
              className="pf-l-grid__item"
            >
              <GridItem
                className=""
                key="firstName"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={true}
                    component={[Function]}
                    format={[Function]}
                    isRequired={false}
                    label="First name"
                    name="firstName"
                    parse={[Function]}
                    type="text"
                    validate={[Function]}
                  >
                    <TextField
                      autoFocus={true}
                      input={
                        Object {
                          "name": "firstName",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      isRequired={false}
                      label="First name"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": "Required",
                          "initial": undefined,
                          "invalid": true,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": false,
                          "visited": false,
                        }
                      }
                      type="text"
                    >
                      <FinalFormField
                        autoFocus={true}
                        componentType="textField"
                        input={
                          Object {
                            "name": "firstName",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="First name"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": "Required",
                            "initial": undefined,
                            "invalid": true,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": false,
                            "visited": false,
                          }
                        }
                        type="text"
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            First name
                          </label>
                          <TextInput
                            aria-label={null}
                            autoFocus={true}
                            className=""
                            id="firstName"
                            isAlt={false}
                            isDisabled={false}
                            isReadOnly={false}
                            isRequired={false}
                            isValid={true}
                            name="firstName"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={true}
                              className="pf-c-form-control"
                              disabled={false}
                              id="firstName"
                              name="firstName"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              readOnly={false}
                              required={false}
                              type="text"
                              value=""
                            />
                          </TextInput>
                        </div>
                      </FinalFormField>
                    </TextField>
                  </Field>
                </div>
              </GridItem>
              <GridItem
                className=""
                key="lastName"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={false}
                    component={[Function]}
                    format={[Function]}
                    isRequired={false}
                    label="Last name"
                    name="lastName"
                    parse={[Function]}
                    type="text"
                    validate={[Function]}
                  >
                    <TextField
                      autoFocus={false}
                      input={
                        Object {
                          "name": "lastName",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      isRequired={false}
                      label="Last name"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": "Required",
                          "initial": undefined,
                          "invalid": true,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": false,
                          "visited": false,
                        }
                      }
                      type="text"
                    >
                      <FinalFormField
                        autoFocus={false}
                        componentType="textField"
                        input={
                          Object {
                            "name": "lastName",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="Last name"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": "Required",
                            "initial": undefined,
                            "invalid": true,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": false,
                            "visited": false,
                          }
                        }
                        type="text"
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            Last name
                          </label>
                          <TextInput
                            aria-label={null}
                            autoFocus={false}
                            className=""
                            id="lastName"
                            isAlt={false}
                            isDisabled={false}
                            isReadOnly={false}
                            isRequired={false}
                            isValid={true}
                            name="lastName"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={false}
                              className="pf-c-form-control"
                              disabled={false}
                              id="lastName"
                              name="lastName"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              readOnly={false}
                              required={false}
                              type="text"
                              value=""
                            />
                          </TextInput>
                        </div>
                      </FinalFormField>
                    </TextField>
                  </Field>
                </div>
              </GridItem>
              <GridItem
                className=""
                key="age"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={false}
                    component={[Function]}
                    format={[Function]}
                    label="Age"
                    name="age"
                    parse={[Function]}
                    type="number"
                    validate={[Function]}
                  >
                    <TextField
                      autoFocus={false}
                      input={
                        Object {
                          "name": "age",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      isRequired={false}
                      label="Age"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": undefined,
                          "initial": undefined,
                          "invalid": false,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": true,
                          "visited": false,
                        }
                      }
                      type="number"
                    >
                      <FinalFormField
                        autoFocus={false}
                        componentType="textField"
                        input={
                          Object {
                            "name": "age",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="Age"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": undefined,
                            "initial": undefined,
                            "invalid": false,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": true,
                            "visited": false,
                          }
                        }
                        type="number"
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            Age
                          </label>
                          <TextInput
                            aria-label={null}
                            autoFocus={false}
                            className=""
                            id="age"
                            isAlt={false}
                            isDisabled={false}
                            isReadOnly={false}
                            isRequired={false}
                            isValid={true}
                            name="age"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="number"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={false}
                              className="pf-c-form-control"
                              disabled={false}
                              id="age"
                              name="age"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              readOnly={false}
                              required={false}
                              type="number"
                              value=""
                            />
                          </TextInput>
                        </div>
                      </FinalFormField>
                    </TextField>
                  </Field>
                </div>
              </GridItem>
              <GridItem
                className=""
                key="bio"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={false}
                    component={[Function]}
                    format={[Function]}
                    label="Bio"
                    name="bio"
                    parse={[Function]}
                    validate={[Function]}
                  >
                    <TextareaField
                      autoFocus={false}
                      input={
                        Object {
                          "name": "bio",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      label="Bio"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": undefined,
                          "initial": undefined,
                          "invalid": false,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": true,
                          "visited": false,
                        }
                      }
                    >
                      <FinalFormField
                        autoFocus={false}
                        componentType="textarea"
                        input={
                          Object {
                            "name": "bio",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="Bio"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": undefined,
                            "initial": undefined,
                            "invalid": false,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": true,
                            "visited": false,
                          }
                        }
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            Bio
                          </label>
                          <TextArea
                            aria-label={null}
                            autoFocus={false}
                            className=""
                            id="bio"
                            isRequired={false}
                            isValid={true}
                            name="bio"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            value=""
                          >
                            <textarea
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={false}
                              className="pf-c-form-control"
                              id="bio"
                              name="bio"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              required={false}
                              value=""
                            />
                          </TextArea>
                        </div>
                      </FinalFormField>
                    </TextareaField>
                  </Field>
                </div>
              </GridItem>
              <GridItem
                className=""
                key="password"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={false}
                    component={[Function]}
                    format={[Function]}
                    label="Password"
                    name="password"
                    parse={[Function]}
                    type="password"
                    validate={[Function]}
                  >
                    <TextField
                      autoFocus={false}
                      input={
                        Object {
                          "name": "password",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      isRequired={false}
                      label="Password"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": undefined,
                          "initial": undefined,
                          "invalid": false,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": true,
                          "visited": false,
                        }
                      }
                      type="password"
                    >
                      <FinalFormField
                        autoFocus={false}
                        componentType="textField"
                        input={
                          Object {
                            "name": "password",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="Password"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": undefined,
                            "initial": undefined,
                            "invalid": false,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": true,
                            "visited": false,
                          }
                        }
                        type="password"
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            Password
                          </label>
                          <TextInput
                            aria-label={null}
                            autoFocus={false}
                            className=""
                            id="password"
                            isAlt={false}
                            isDisabled={false}
                            isReadOnly={false}
                            isRequired={false}
                            isValid={true}
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="password"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={false}
                              className="pf-c-form-control"
                              disabled={false}
                              id="password"
                              name="password"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              readOnly={false}
                              required={false}
                              type="password"
                              value=""
                            />
                          </TextInput>
                        </div>
                      </FinalFormField>
                    </TextField>
                  </Field>
                </div>
              </GridItem>
              <GridItem
                className=""
                key="telephone"
                lg={null}
                lgOffset={null}
                lgRowSpan={null}
                md={null}
                mdOffset={null}
                mdRowSpan={null}
                offset={null}
                rowSpan={null}
                sm={12}
                smOffset={null}
                smRowSpan={null}
                span={null}
                xl={null}
                xlOffset={null}
                xlRowSpan={null}
              >
                <div
                  className="pf-l-grid__item pf-m-12-col-on-sm"
                >
                  <Field
                    autoFocus={false}
                    component={[Function]}
                    format={[Function]}
                    label="Telephone"
                    name="telephone"
                    parse={[Function]}
                    type="text"
                    validate={[Function]}
                  >
                    <TextField
                      autoFocus={false}
                      input={
                        Object {
                          "name": "telephone",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "onFocus": [Function],
                          "value": "",
                        }
                      }
                      isRequired={false}
                      label="Telephone"
                      meta={
                        Object {
                          "active": false,
                          "data": Object {},
                          "dirty": false,
                          "dirtySinceLastSubmit": false,
                          "error": undefined,
                          "initial": undefined,
                          "invalid": false,
                          "pristine": true,
                          "submitError": undefined,
                          "submitFailed": false,
                          "submitSucceeded": false,
                          "touched": false,
                          "valid": true,
                          "visited": false,
                        }
                      }
                      type="text"
                    >
                      <FinalFormField
                        autoFocus={false}
                        componentType="textField"
                        input={
                          Object {
                            "name": "telephone",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "onFocus": [Function],
                            "value": "",
                          }
                        }
                        isRequired={false}
                        label="Telephone"
                        meta={
                          Object {
                            "active": false,
                            "data": Object {},
                            "dirty": false,
                            "dirtySinceLastSubmit": false,
                            "error": undefined,
                            "initial": undefined,
                            "invalid": false,
                            "pristine": true,
                            "submitError": undefined,
                            "submitFailed": false,
                            "submitSucceeded": false,
                            "touched": false,
                            "valid": true,
                            "visited": false,
                          }
                        }
                        type="text"
                      >
                        <div
                          className="pf-c-form__group"
                        >
                          <label
                            className="pf-c-form__label "
                          >
                            Telephone
                          </label>
                          <TextInput
                            aria-label={null}
                            autoFocus={false}
                            className=""
                            id="telephone"
                            isAlt={false}
                            isDisabled={false}
                            isReadOnly={false}
                            isRequired={false}
                            isValid={true}
                            name="telephone"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              autoFocus={false}
                              className="pf-c-form-control"
                              disabled={false}
                              id="telephone"
                              name="telephone"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              readOnly={false}
                              required={false}
                              type="text"
                              value=""
                            />
                          </TextInput>
                        </div>
                      </FinalFormField>
                    </TextField>
                  </Field>
                </div>
              </GridItem>
            </div>
          </GridItem>
          <GridItem
            className=""
            lg={null}
            lgOffset={null}
            lgRowSpan={null}
            md={null}
            mdOffset={null}
            mdRowSpan={null}
            offset={null}
            rowSpan={null}
            sm={null}
            smOffset={null}
            smRowSpan={null}
            span={null}
            xl={null}
            xlOffset={null}
            xlRowSpan={null}
          >
            <div
              className="pf-l-grid__item"
            >
              <Button
                aria-label={null}
                className=""
                component="button"
                id="form-renderer-submit"
                isActive={false}
                isBlock={false}
                isDisabled={false}
                isFocus={false}
                isHover={false}
                onClick={[Function]}
                type="button"
                variant="primary"
              >
                <button
                  aria-disabled={null}
                  aria-label={null}
                  className="pf-c-button pf-m-primary"
                  disabled={false}
                  id="form-renderer-submit"
                  onClick={[Function]}
                  tabIndex={null}
                  type="button"
                >
                  Submit
                </button>
              </Button>
            </div>
          </GridItem>
        </div>
      </Grid>
    </form>
  </ReactFinalForm>
</FormRenderer>
`;
